modules/geochemistry/include/utils/GeochemicalSystem.h

```diff
@@ -0,0 +1,934 @@
+   * @param constraint_meaning A list the provides physical meaning to the constraint_values
+   * @param initial_temperature Initial temperature
+   * @param iters_to_make_consistent The initial equilibrium molalities depend on the activity
+  coefficients, which depend on the basis and equilibrium molalities.  This circular dependence
+  means it is usually impossible to define an exactly consistent initial configuration for
+  molalities.  An iterative process is used to approach the consistent initial configuration, using
+  iters_to_make_consistent iterations.  Usually iters_to_make_consistent=0 is reasonable, because
+  there are so many approximations used in the solution process that a slightly inconsistent initial
+  configuration is fine
+   * @param min_initial_molality Minimum value of equilibrium molality used in the initial condition
+   * @param kin_name Names of the kinetic species that are provided with initial conditions in
+  kin_initial_moles.  All kinetic species must be provided with an initial condition
+   * @param kin_initial_moles values of the initial mole number for the kinetic species
+   */
+  GeochemicalSystem(ModelGeochemicalDatabase & mgd,
+                    GeochemistryActivityCoefficients & gac,
+                    const GeochemistryIonicStrength & is,
+                    GeochemistrySpeciesSwapper & swapper,
+                    const std::vector<std::string> & swap_out_of_basis,
+                    const std::vector<std::string> & swap_into_basis,
+                    const std::string & charge_balance_species,
+                    const std::vector<std::string> & constrained_species,
+                    const std::vector<Real> & constraint_value,
+                    const MultiMooseEnum & constraint_meaning,
+                    Real initial_temperature,
+                    unsigned iters_to_make_consistent,
+                    Real min_initial_molality,
+                    const std::vector<std::string> & kin_name,
+                    const std::vector<Real> & kin_initial_moles);
+
+  GeochemicalSystem(ModelGeochemicalDatabase & mgd,
+                    GeochemistryActivityCoefficients & gac,
+                    const GeochemistryIonicStrength & is,
+                    GeochemistrySpeciesSwapper & swapper,
+                    const std::vector<std::string> & swap_out_of_basis,
+                    const std::vector<std::string> & swap_into_basis,
+                    const std::string & charge_balance_species,
+                    const std::vector<std::string> & constrained_species,
+                    const std::vector<Real> & constraint_value,
+                    const std::vector<ConstraintMeaningEnum> & cc,
+                    Real initial_temperature,
+                    unsigned iters_to_make_consistent,
+                    Real min_initial_molality,
+                    const std::vector<std::string> & kin_name,
+                    const std::vector<Real> & kin_initial_moles);
```

Comment:
You've got a _ton_ of code duplication here in constructors but I can't seem to find a way to make it much better!